snmp/datadog_checks/snmp/snmp.py

```diff
@@ -83,6 +87,10 @@ def __init__(self, name, init_config, instances):
             self.profiles,
             self.profiles_by_oid,
         )
+        if self._config.network_address:
+            self._thread = threading.Thread(target=self.discover_instances)
```

Comment:
(minor) Could help debugging by setting the Thread name using the `self.check_id` (not sure if it's available at this stage) or `name` for example.

Reply:
Yeah check_id is not, used name.